Cart context and Checkout

AndreRimes opened the pull request:
- Resolution of Issue #50
    - Creation of the cart context along with the local storage logic.
    - Integration of this context with the cart page/cart product.
    - Add an "Add to Cart" button on the main page, just so I can add something to the cart. It is not on Figma, and if you guys don't like it, feel free to remove it.
- Resolution of Issue #45
    - Creating the checkout modal.
    -API route '/api/sendEmail' that sends an email to the user with a verification code and creates a purchase in the database.
    -Creation of the confirmation component where the user can input the verification code or cancel the purchase.
    - API route '/api/emailConfirmation' that receives the code sent by the user and confirms if it is correct and matches the one in the database.
    - API route '/api/deleteEmail/[id]' for deleting a purchase.
    - Components for loading, error, and success to give users feedback. It's worth noting that the design for the success and error components wasn't in Figma, so we may need to improve the design.

AndreRimes committed:
Checkout

add checkout modal

AndreRimes committed:
Merge branch 'main' into Checkout-Modal

AndreRimes committed:
API rout to send emails

-Creation of api rout to send emails with node mailer
-Loading/sucess/error changes in the modal

AndreRimes committed:
cartContext

- creation of the context
-local storage

AndreRimes committed:
Finalization of the checkout logic

Finalization of the checkout logic

AndreRimes committed:
remove the test smtp

remove the test smtp

AndreRimes committed:
env in the email auth

Using environment variables in the specification to call the email:

AndreRimes committed:
resolving pull request warnings

Ajknight121 commented on app/components/animationFunction.js:
Can you instead define a scaleup animation in the tailwind config. That could then be applied on hover:: to components

Ajknight121 commented on app/api/sendEmail/route.js:
```diff
@@ -0,0 +1,91 @@
+    let cartItemsHtml = '';
+    for (const item of cart) {
+        if (item.cartQuantity > item.quantity) {
+            return NextResponse.json('Out of stock', { status: 500 });
```
Add all items that are out of stock to the message ex:
"Out of Stock:
    Item1
    Item2"
 We will have to be created at a later time.

Ajknight121 commented on app/components/modalCheckout.jsx:
```diff
@@ -0,0 +1,135 @@
+
+        const email = emailRef.current.value
+        const name = nameRef.current.value
+
```
A more complex error check is needed, all present errors should be shown on submit and errors should go away when corrected, at least after submit.

Ajknight121 commented on app/components/product.jsx:
```diff
@@ -1,30 +1,53 @@
-export default function Product({ id, title, price, imgUrl,isDarkMode}) {
+
+export default function Product({ product, isDarkMode }) {
   const [hovered, setHovered] = useState(false);
```
Tailwind has the hover:: property. adding hover::styling will apply styles when the component is hovered over.

Ajknight121 commented on app/components/modalCheckout.jsx:
More error checking needed in this file

Ajknight121 reviewed (approved):
Excellent work! There are a few more cleanups but those can be made in future pulls.

Ajknight121 committed:
Merge branch 'main' into CartContext

AndreRimes committed:
Changes requested to pull

Add functions to handle email and name input fields that check if the email or name matches a specified pattern when there is a change in the input fields.

Implement a hover effect for buttons using the Tailwind

List all items that are currently out of stock, if any.

AndreRimes committed:
Remove apostrophe Error

Remove apostrophe Error
